docs(POD-036): README.md (install + quickstart + format + catalogue) (#23)
invaderDMG/podcast-script: README.md

implicit grep contract for shell wrappers.

| Group | Tokens |
|-------|--------|
| Lifecycle (4) | `startup`, `config_loaded`, `backend_selected`, `done` |
| Model lifecycle (3) | `model_load_start`, `model_load_done`, `model_download` |
| Phase boundaries (8) | `decode_start`, `decode_done`, `segment_start`, `segment_done`, `transcribe_start`, `transcribe_done`, `render_done`, `write_done` |
| Terminal errors (6) | `usage_error`, `input_io_error`, `decode_error`, `model_error`, `output_exists`, `internal_error` |

`event=done` is the locked summary line per SRS UC-1 step 10. Its key
order is preserved by the formatter:

```
level=info event=done input=<path> output=<path> backend=<name> model=<name> lang=<code> duration_in_s=<n> duration_wall_s=<n>
```

`event=model_download` (UC-2 step 3) carries `model=…` and a `size_gb=…`
hint so you can decide whether to step away from the keyboard for a
moment. A future addition to the catalogue is a SemVer-minor change;
removing or renaming an existing token is a SemVer-major break (per
SRS §16.1).

The full ADR is at
[`docs/adr/0012-event-catalogue-freeze.md`](docs/adr/0012-event-catalogue-freeze.md).

> ### Verbosity matrix (AC-US-3.5)

Author: **[suggestion]** [docs] — Verbosity matrix is unrelated to the event catalogue but lives as `###` under it

The `## Logfmt event catalogue` section ends with the `event=done` shape; making the verbosity matrix a `###` under it implies the matrix is part of the catalogue, but they're separate concerns (logger level + bar visibility, AC-US-3.5).

Proposed fix:

```suggestion
## Verbosity matrix (AC-US-3.5)
```

Feel free to ignore — purely a structure/navigation nit.

| Flag | Logger level | Progress bar | Notes |
|------|--------------|--------------|-------|
| `-q` / `--quiet` | `ERROR` | Suppressed | No `event=*` info lines on the happy path; only error lines if any |
| _(default)_ | `INFO` | On (TTY) / off (non-TTY) | The "normal" surface |
| `-v` / `--verbose` | `DEBUG` | On (TTY) / off (non-TTY) | Reserved — no debug-level events emitted in v1.0 |

> | `--debug` | `DEBUG` | On (TTY) / off (non-TTY) | Same as `-v` plus the artifact directory of US-7 |

Author: **[should-fix]** [docs] — Progress-bar column says "off (non-TTY)", contradicting AC-US-3.2

`SRS.md` AC-US-3.2 (line 249): *"the progress bar **degrades to a one-line-per-phase plain-text update** and does not emit ANSI control sequences."* It is not turned off — it falls back to logfmt-shaped per-phase lines.

The README's own Quickstart at line 60 already says it correctly: *"a [rich] progress bar on a TTY — degraded to plain logfmt on a pipe (AC-US-3.2)."* The matrix three sections later contradicts that.

Cited rule: `SRS.md:249` (AC-US-3.2).

Proposed fix — replace the three matching cells:

```suggestion
| _(default)_ | `INFO` | On (TTY) / plain-text fallback (non-TTY) | The "normal" surface |
| `-v` / `--verbose` | `DEBUG` | On (TTY) / plain-text fallback (non-TTY) | Reserved — no debug-level events emitted in v1.0 |
| `--debug` | `DEBUG` | On (TTY) / plain-text fallback (non-TTY) | Same as `-v` plus the artifact directory of US-7 |
```

The three flags are mutually exclusive (SRS §9.1 `[-v | -q | --debug]`);
combining any two exits 2.

## Observed throughput (no v1 SLA)

The project doesn't commit to a wall-clock target (NFR-1). Observed
end-to-end timings on the maintainer's reference machines, on the
bundled 60-second fixture:

| Backend / model | Hardware | First run (cold cache) | Steady state |
|-----------------|----------|------------------------|--------------|
| `faster-whisper` `tiny` | macOS Apple Silicon | ~30 s (~75 MB download) | ~7 s |
| `faster-whisper` `tiny` | Ubuntu CI runner (CPU) | ~30 s | ~10 s |
| `mlx-whisper` `tiny` | macOS Apple Silicon | (download then) ~5 s | ~5 s |

Numbers refresh whenever a release is cut. If your throughput differs
substantially, file an issue with the `event=done` summary line and the
output of `uname -a` — see Risk #7 in the project plan.

## Development

> The project follows a tier-3 test pyramid (ADR-0017):

Author: **[should-fix]** [docs] — "tier-3 test pyramid" misrepresents ADR-0017

ADR-0017 §Decision (`docs/adr/0017-three-tier-test-strategy.md:15-60`) commits to **three** tiers under `tests/{unit,contract,integration}/`:

- **Tier 1 — unit** (no heavy deps, ≤ 1 s)
- **Tier 2 — contract** (the parametrized `WhisperBackend` Protocol invariants — the whole reason ADR-0017 supersedes the original two-tier plan)
- **Tier 3 — integration** (full CLI on `examples/sample.mp3`)

The Development section shows only the default tier and `pytest -m slow` (which corresponds to Tier 3). Tier 2 is silently dropped, and "tier-3 test pyramid" reads as "the pyramid that includes tier 3," not as "three-tier strategy."

Cited rule: `docs/adr/0017-three-tier-test-strategy.md` §Decision (Accepted).

Proposed fix — pick one:

```suggestion
The project follows a three-tier test strategy (ADR-0017) — unit, contract, integration:

```sh
uv run pytest                     # tiers 1 + 2 (unit + contract), default
uv run pytest -m slow             # tier 3 (real CLI on examples/sample.mp3)
uv run pytest -m "slow or not slow"   # everything (CI does this)
```
```

Or, if the `tests/contract/` directory hasn't been split out yet from the current flat `tests/` layout (the repo state suggests this), keep the simpler example but add a one-liner caveat under the code block: *"Tier 2 (contract) tests live alongside the unit tier in `tests/` for now; ADR-0017's `tests/contract/` split is tracked separately."* Either is fine — the current text overpromises the live tree.

```sh
uv run pytest                     # tier 1 (unit + fakes), default
uv run pytest -m slow             # tier 3 (real CLI on examples/sample.mp3)
uv run pytest -m "slow or not slow"   # everything (CI does this)

uv run ruff check .               # lint
uv run ruff format --check .      # format
uv run mypy --strict src tests    # type check
```

The slow tier downloads the `tiny` model on first run (~75 MB). A
warning-as-error pytest policy is in effect — any unhandled Python
warning fails the suite (NFR-10 stderr cleanliness).

The architecture lives in
[`SYSTEM_DESIGN.md`](SYSTEM_DESIGN.md) and the 17 accepted ADRs under
[`docs/adr/`](docs/adr/); the requirements are in
[`SRS.md`](SRS.md); the sprint plan in
[`PROJECT_PLAN.md`](PROJECT_PLAN.md).

## License

> MIT — see `pyproject.toml` for the canonical declaration.

Author: **[blocking]** [docs] — License declared by README, but no canonical declaration exists

This line tells the reader the canonical license declaration lives in `pyproject.toml`. It doesn't:

- `pyproject.toml` (lines 1-22 of `[project]`) has no `license` field, no `license-files`, no SPDX identifier.
- There is no `LICENSE` (or `LICENSE.md`, `COPYING`) file at the repo root.

For a public repo on the v1.0.0 path (`PROJECT_PLAN.md` §16 + `PROJECT_BRIEF.md` §14), this matters: under default US copyright the project is *all rights reserved* until a license is actually declared, which contradicts the README's MIT claim and would make external use legally ambiguous.

Cited rule: `PROJECT_BRIEF.md` §14 (license discipline) + the PR body's own claim that the README is publishable for SP-7's dogfooders.

Proposed fix — **two edits, both required**:

1. Add to `pyproject.toml` `[project]` (PEP 639):

   ```toml
   license = "MIT"
   license-files = ["LICENSE"]
   ```

2. Add a `LICENSE` file at the repo root with the MIT text and the maintainer's copyright line (`Copyright (c) 2026 Juan García`).

Alternatively, if the project is **not** intended to be MIT, change this README line to match whatever license is actually being declared — but the README and `pyproject.toml` must agree, and one of them must be the source of truth that actually exists.
